tests/test_comparison_template_lib.py

```diff
@@ -0,0 +1,146 @@
+import pandas as pd
+import pytest
+
+import splink.duckdb.duckdb_comparison_library as cld
+import splink.duckdb.duckdb_comparison_template_library as ctld
+import splink.spark.spark_comparison_library as cls
+import splink.spark.spark_comparison_template_library as ctls
+
+from splink.duckdb.duckdb_linker import DuckDBLinker
+from splink.spark.spark_linker import SparkLinker
+
+
+def test_duckdb_date_comparison_run():
+    ctld.date_comparison("date")
+
+
+def test_spark_date_comparison_run():
+    ctls.date_comparison("date")
+
+
+def test_datediff_levels(spark):
+    df = pd.DataFrame(
+        [
+            {
+                "unique_id": 1,
+                "first_name": "Tom",
+                "dob": "2000-01-01",
+            },
+            {
+                "unique_id": 2,
+                "first_name": "Robin",
+                "dob": "2000-01-30",
+            },
+            {
+                "unique_id": 3,
+                "first_name": "Zoe",
+                "dob": "1995-09-30",
+            },
+            {
+                "unique_id": 4,
+                "first_name": "Sam",
+                "dob": "1966-07-30",
+            },
+            {
+                "unique_id": 5,
+                "first_name": "Andy",
+                "dob": "1996-03-25",
+            },
+            {
+                "unique_id": 6,
+                "first_name": "Alice",
+                "dob": "2000-03-25",
+            },
+            {
+                "unique_id": 7,
+                "first_name": "Afua",
+                "dob": "1960-01-01",
+            },
+        ]
+    )
+
+    # Generate our various settings objs
+    duck_settings = {
+        "link_type": "dedupe_only",
+        "comparisons": [ctld.date_comparison("dob")],
+    }
+
+    print(duck_settings)
+
+    spark_settings = {
+        "link_type": "dedupe_only",
+        "comparisons": [ctld.date_comparison("dob")],
+    }
+
+    # We need to put our column in datetime format for this to work
+    df["dob"] = pd.to_datetime(df["dob"])
+
+    # DuckDBFrame
+    dlinker = DuckDBLinker(df, duck_settings)
+    duck_df_e = dlinker.predict().as_pandas_dataframe()
+
+    # SparkFrame
+    sparkdf = spark.createDataFrame(df)
+    sparkdf.persist()
+
+    splinker = SparkLinker(
+        sparkdf,
+        spark_settings,
+    )
+    sp_df_e = splinker.predict().as_pandas_dataframe()
+
+    # # Dict key: {size: gamma_level value}
+    size_gamma_lookup = {0: 17, 1: 3, 2: 0, 3: 1}
+
+    linker_outputs = {
+        "dlinker": duck_df_e  # ,
+        # "slinker": sp_df_e,
```

Comment:
Do you want this comment out?

Reply:
Unless this is the failing test that you were referring to?

Reply:
Yes, this is the failing test

Reply:
No worries, I'll take a look now.

Reply:
We're getting different results for spark vs duckdb.

I'm just checking why now